File: Solutions/FirstProgram.f90
PROGRAM MAIN

  USE omp_lib

  INTEGER, PARAMETER :: T=10000
  INTEGER :: sum
  INTEGER :: i

  sum = 0
  
  ! Simple accumulation loop
  ! (Note in Fortran we can use a built-in for this, but
  ! ignore that for this demo)

!$OMP PARALLEL DO REDUCTION(+:sum)
  DO i = 1, T
    sum = sum + i
  END DO
!$OMP END PARALLEL DO 

  PRINT*, "Loop gave        ", sum

  PRINT*, "Answer should be ", T*(T+1)/2

END PROGRAM

File: Solutions/FirstProgram.c
#include<stdio.h>
#include<omp.h>

int main(){

  const int T = 10000;
  int sum;

  sum = 0;
  
  // Simple accumulation loop
  // (Note in Fortran we can use a built-in for this, but
  // ignore that for this demo)
#pragma omp parallel for reduction(+:sum)
  for(int i = 1; i <= T; i++){
    sum = sum + i;
  }

  printf("Loop gave         %i\n", sum);
  printf("Answer should be  %i\n", T*(T+1)/2);

}
